### PR TITLE
python3Packages.diofant: 0.10.0 -> 0.12.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.12.0

Change log: https://diofant.readthedocs.io/en/latest/release/notes-0.12.html

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nlewo** [comment]
There are two changes that seems to be unrelated. Is it a mistake?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129041` at 6a6601f5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>python38Packages.datrie</li>
    <li>python38Packages.diofant</li>
    <li>python39Packages.datrie</li>
    <li>python39Packages.diofant</li>
    <li>snakemake</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/datrie/default.nix:4:3:
```
  |
4 | , fetchpatch
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/diofant/default.nix:39:0:
```
   |
39 |     description = "A Python CAS library";
   | ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
> There are two changes that seems to be unrelated. Is it a mistake?

Yes, the `datrie` update is misplaced.

**Comment by risicle** [comment]
Would be nice to have at least a `pythonImportsCheck` for this.

But builds happily, macos 10.15 & nixos x86_64.

**Comment by fabaff** [comment]
> Would be nice to have at least a pythonImportsCheck for this.

Indeed, added.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129041` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python39Packages.diofant</li>
  </ul>
</details>

File: pkgs/development/python-modules/diofant/default.nix [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
     inherit version;
     pname = "Diofant";
-    sha256 = "0qjg0mmz2cqxryr610mppx3virf1gslzrsk24304502588z53v8w";
+    sha256 = "sha256-G0CTSoDSduiWxlrk5XjnX5ldNZ9f7yxaJeUPO3ezJgo=";
   };
 
   nativeBuildInputs = [
     isort
-    pytestrunner
+    pytest-runner
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/diofant/default.nix:28]: Can we patch this out?

Comment (fabaff) [reply pkgs/development/python-modules/diofant/default.nix]: That was removed. Thanks for checking.

**Pull request merged**